app/segments/(team)/chooseTeam.js

```diff
@@ -58,9 +63,44 @@ function ChooseTeam() {
     }
   }, [currentUserId]);
 
-  if (loading) {
-    return <Loading />;
-  }
+  useEffect(() => {
+    const unregisterAuthObserver = onIdTokenChanged(auth, async (user) => {
+      if (user) {
+        if (user.emailVerified) {
+          setVerified(true);
+          showSnackBar("Email successfully verified.");
+          clearInterval(intervalId); // Stop the interval when email is verified
+          unregisterAuthObserver(); // Unregister the auth observer
+        } else {
+          setVerified(false);
+          console.log("Error: Email Not Verified Yet, Try Again");
+        }
+      }
+    });
+
+    // Set up an interval to check email verification every 10 seconds
+    const intervalId = setInterval(async () => {
+      if (auth.currentUser) {
+        await auth.currentUser.reload();
+      }
+    }, 10000); // 10,000 ms = 10 seconds
+
+    return () => {
+      clearInterval(intervalId); // Clean up the interval when component unmounts
+      unregisterAuthObserver(); // Unregister the auth observer
+    };
+  }, []);
+
+  const [refreshing, setRefreshing] = useState(false);
+
+  const onRefresh = useCallback(() => {
```

Comment:
Is this for debouncing? This looks like a debouncing attempt to me, but it only delay the reload everytime the user hit refresh. If that's the case, i don't think we need it.

Reply:
I removed the setTImeout from this, which is hopefully what you meant.